### PR TITLE
[CDAP-14778] Fix logic to determine what namespaces to search metadata

### PR DESCRIPTION
**Pull request by anew**
This fixes the logic that generates the namespace-qualified search terms from a query.
Especially the case where the search is on namespace "system" and the requested entity scope is also "system".

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java [hunk 1]
```diff
@@ -829,22 +830,47 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
...
+    }
+    return searchTerms;
+  }
+
+  @VisibleForTesting
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  static Consumer<String> determineSearchFields(Optional<NamespaceId> namespace,
+                                                Set<EntityScope> entityScopes, List<SearchTerm> searchTerms) {
+    if (!namespace.isPresent()) {
+      // we can't really represent "any namespace but system", so we just search for any occurence of the term
```
Comment (rohitsinha54) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java]: typo: occurrence

**Review by rohitsinha54 — APPROVED** [review]
Just one nit comment.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java (hunk 1 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java]: thx - fixed.

**Comment by anew** [comment]
Fixed typo and squashed. Build passed: https://builds.cask.co/browse/CDAP-DUT6752/latest. Merging now.

**Pull request merged**